gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/spi/SubscriberApi.java

```diff
@@ -0,0 +1,663 @@
+  public static final PathTemplate SUBSCRIPTION_PATH_TEMPLATE =
+      PathTemplate.create("/projects/{project}/subscriptions/{subscription}");
+
+  // ========
+  // Members
+  // ========
+
+  private final ManagedChannel channel;
+  private final ServiceApiSettings settings;
+
+  // ===============
+  // Factory Methods
+  // ===============
+
+  /**
+   * Constructs an instance of SubscriberApi with default settings.
+   */
+  public static SubscriberApi create() throws IOException {
+    return create(new ServiceApiSettings());
+  }
+
+  /**
+   * Constructs an instance of SubscriberApi, using the given settings. The channels are created based
+   * on the settings passed in, or defaults for any settings that are not set.
+   */
+  public static SubscriberApi create(ServiceApiSettings settings) throws IOException {
+    return new SubscriberApi(settings);
+  }
+
+  private SubscriberApi(ServiceApiSettings settings) throws IOException {
+    ServiceApiSettings internalSettings = ApiUtils.settingsWithChannels(settings,
+        SERVICE_ADDRESS, DEFAULT_SERVICE_PORT, ALL_SCOPES);
+    this.settings = internalSettings;
+    this.channel = internalSettings.getChannel();
+  }
+
+  // ==============================
+  // Resource Name Helper Functions
+  // ==============================
+
+  public static final String createProject(String project) {
+    return PROJECT_PATH_TEMPLATE.instantiate(
+        "project", project);
+  }
+
+  public static final String createSubscription(String project, String subscription) {
+    return SUBSCRIPTION_PATH_TEMPLATE.instantiate(
+        "project", project,"subscription", subscription);
+  }
+
+
+  // ========
+  // Getters
+  // ========
+
+  public Channel getChannel() {
+    return channel;
+  }
+
+
+  // =============
+  // Service Calls
+  // =============
+
+  // ----- createSubscription -----
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Creates a subscription to a given topic for a given subscriber.
+   * If the subscription already exists, returns ALREADY_EXISTS.
+   * If the corresponding topic doesn't exist, returns NOT_FOUND.
+   *
+   * If the name is not provided in the request, the server will assign a random
+   * name for this subscription on the same project as the topic.
+   *
+   * @param name The name of the subscription. It must have the format
+   * `"projects/{project}/subscriptions/{subscription}"`. `{subscription}` must
+   * start with a letter, and contain only letters (`[A-Za-z]`), numbers
+   * (`[0-9]`), dashes (`-`), underscores (`_`), periods (`.`), tildes (`~`),
+   * plus (`+`) or percent signs (`%`). It must be between 3 and 255 characters
+   * in length, and it must not start with `"goog"`.
+   * @param topic The name of the topic from which this subscription is receiving messages.
+   * The value of this field will be `_deleted-topic_` if the topic has been
```

Comment:
Second part is covered by #385 Do you plan to create a separate issue for the first part?